File: InJava/app/src/main/java/com/barryzeha/quizquestion/model/RankingModel.java
package com.barryzeha.quizquestion.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 01/10/2022
 * Copyright (c)  All rights reserved.
 ***/
@Entity(tableName = "ranking_table")
public class RankingModel {
    @PrimaryKey(autoGenerate = true) private int id;
    private String namePlayer;
    private int points;

    public RankingModel(String namePlayer, int points){
        this.id=0;
        this.namePlayer= namePlayer;
        this.points=points;
    }
    public int getId(){
        return this.id;
    }
    public void setId(int id){
        this.id=id;
    }
    public String getNamePlayer() {
        return namePlayer;
    }

    public void setNamePlayer(String namePlayer) {
        this.namePlayer = namePlayer;
    }

    public int getPoints() {
        return points;
    }

    public void setPoints(int points) {
        this.points = points;
    }
}

File: InKotlin/app/src/main/java/com/barryzeha/quizquestion/model/RankingModel.kt
package com.barryzeha.quizquestion.model

import androidx.room.Entity
import androidx.room.PrimaryKey


/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 27/09/2022
 * Copyright (c)  All rights reserved.
 ***/

@Entity(tableName = "ranking_table")
data class RankingModel (
    @PrimaryKey(autoGenerate = true)var id:Int=0,
    val name:String?="",
    val points:Int
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as RankingModel

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id
    }
}